activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -844,11 +844,27 @@ def configure_connection
 
         defaults = [':default', :default].to_set
 
-        # Make MySQL reject illegal values rather than truncating or blanking them, see
-        # http://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_strict_all_tables
+        # The default SQL mode in MySQL 5.7 includes these modes:
+        #   ONLY_FULL_GROUP_BY, STRICT_TRANS_TABLES, NO_ZERO_IN_DATE, NO_ZERO_DATE,
+        #   ERROR_FOR_DIVISION_BY_ZERO, NO_AUTO_CREATE_USER, and NO_ENGINE_SUBSTITUTION.
+        # See https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sql-mode-setting
```

Comment:
MariaDB is a bit different, defaulting to `NO_ENGINE_SUBSTITUTION,NO_AUTO_CREATE_USER`.